### DISCUSSION TITLE
why downgrade from 3.5 to 3.4 isn't possible?

### DISCUSSION THREAD

**Discussion by lavacat** [etcd-io/etcd#14605] [General]
Per [docs](https://etcd.io/docs/v3.5/upgrades/upgrade_3_5/#downgrade) "downgrade from this completed state is not possible".

Wondering what was the reason for this decision? From SRE perspective this is more risky and slows down 3.5 adoption.

Looking at raft.proto, files are the same in 3.4 and 3.5 (with the exception of extra comments and gogoproto options).
In terms of backend state there are 2 new keys in `meta` bucket

> ➜ bbolt keys default.etcd-3.5/member/snap/db meta
> confState
> consistent_index
> term
> ➜ bbolt keys default.etcd-3.4/member/snap/db meta
> consistent_index

and an extra field in member value

> ➜ bbolt get default.etcd-3.5/member/snap/db members 8e9e05c52164694d
> {"id":10276657743932975437,"peerURLs":["http://localhost:2380"],"name":"default"}
> ➜ bbolt get default.etcd-3.4/member/snap/db members 8e9e05c52164694d
> {"id":10276657743932975437,"peerURLs":["http://localhost:2380"],"name":"default","clientURLs":["http://localhost:2379"]}

I've tried hacking the version and doing downgrade locally and it seams to be working fine.

For reference found 2 large related issues (including design docs):
https://github.com/etcd-io/etcd/issues/11716
https://github.com/etcd-io/etcd/issues/13168

**Comment by serathius** [comment]
Etcd doesn't officially support downgrades (planned for v3.6 release). There is no breaking change in the storage layer between v3.5 and v3.4, however there is known issue that will occur when you upgrade back to v3.5 (v3.5 -> v3.4 -> v3.5).
